services/src/main/scala/cromwell/services/metadata/impl/pubsub/PubSubMetadataServiceActor.scala

```diff
@@ -0,0 +1,132 @@
+  val pubSubAppName = serviceConfig.as[Option[String]]("appName").getOrElse("cromwell")
+
+  val pubSubConnection = createPubSubConnection()
+  createTopicAndSubscription().failed foreach { e => throw e }
+
+  def receive = {
+    case action: PutMetadataAction =>
+      publishMessages(action.events).failed foreach { e =>
+        log.error(e, "Failed to post metadata: " + action.events)
+      }
+    case action: PutMetadataActionAndRespond =>
+      publishMessages(action.events) onComplete {
+        case Success(_) => action.replyTo ! MetadataWriteSuccess(action.events)
+        case Failure(e) => action.replyTo ! MetadataWriteFailure(e, action.events)
+      }
+  }
+
+  protected[this] def createPubSubConnection(): GooglePubSubDAO = {
+    implicit val as = context.system
+
+    val googleConfig = GoogleConfiguration(globalConfig)
+
+    // This class requires a service account auth due to the library used
+    val googleAuth = googleConfig.auth(googleAuthName) match {
+      case Valid(a: ServiceAccountMode) => a
+      case Valid(doh) => throw new IllegalArgumentException(s"Unable to configure PubSubMetadataServiceActor: ${doh.name} was not a service account auth")
+      case Invalid(e) => throw new IllegalArgumentException("Unable to configure PubSubMetadataServiceActor: " + e.toList.mkString(", "))
+    }
+
+    val pemFile = googleAuth.fileFormat match {
+      case p: PemFileFormat => p
+      case _ => throw new IllegalArgumentException("Unable to configure PubSubMetadataServiceActor: the service account must supply a PEM file")
+    }
+
+    // The metric name is pretty useless. It's an artifact of the workbench libraries which we're not using
+    new HttpGooglePubSubDAO(pemFile.accountId, pemFile.file, pubSubAppName, googleProject, "cromwell.pubsub")
+  }
+
+  /**
+    * Creates the specified topic, and optionally the subscription if it was defined. We are ignoring the bool return
+    * type from the HttpGooglePubSubDAO return types as it is not considered an error if the topic/subscription already
+    * exist (in fact, this is a likely circumstance).
+    */
+  private def createTopicAndSubscription(): Future[Unit] = {
+    log.info("Ensuring topic " + pubSubTopicName + " exists")
+    pubSubConnection.createTopic(pubSubTopicName) map { _ => createSubscription() } map { _ => () }
+  }
+
+  /**
+    * Creates a subscription if pubsubTopicName is defined. Note that we're ignoring the bool in the Future[bool]
+    * return type of HttpGooglePubSubDAO.createSubscription and only caring about the success of the Future. It is not
+    * considered an error if the subscription already exists (in fact, this is a likely circumstance).
+    */
+  private def createSubscription(): Future[Unit] = {
+    pubSubSubscriptionName match {
+      case Some(name) =>
+        log.info("Creating subscription " + name)
+        pubSubConnection.createSubscription(pubSubTopicName, name) map { _ => () }
```

Comment:
`void`

Reply:
@mcovarr lol I knew you had some little thing on these but I couldn't remember, so I figured your eagle eyes would spot it here